packages/iTwinUI-react/src/core/utils/components/Resizer.tsx

```diff
@@ -0,0 +1,298 @@
+      height: initialHeight,
+    } = elementRef.current.getBoundingClientRect();
+
+    let width = `${initialWidth}px`;
+    let height = `${initialHeight}px`;
+    let translateX = initialTranslateX;
+    let translateY = initialTranslateY;
+
+    const minWidth = Number(
+      getComputedStyle(elementRef.current).minWidth.replace('px', ''),
+    );
+    const minHeight = Number(
+      getComputedStyle(elementRef.current).minHeight.replace('px', ''),
+    );
+
+    const resizer = event.currentTarget.dataset.iuiResizer;
+
+    const originalUserSelect =
+      elementRef.current.ownerDocument.body.style.userSelect;
+    elementRef.current.ownerDocument.body.style.userSelect = 'none';
+
+    const onResizePointerMove = (event: PointerEvent) => {
+      if (!elementRef.current) {
+        return;
+      }
+
+      const containerRect = containerRef?.current?.getBoundingClientRect();
+      const clientX = getBoundedValue(
+        event.clientX,
+        containerRect?.left ?? 0,
+        containerRect?.right ?? getWindow()?.innerWidth ?? 0,
+      );
+      const clientY = getBoundedValue(
+        event.clientY,
+        containerRect?.top ?? 0,
+        containerRect?.bottom ?? getWindow()?.innerHeight ?? 0,
+      );
+
+      switch (resizer) {
+        case 'top-left': {
+          const newHeight = initialHeight + (initialPointerY - clientY);
+          if (newHeight >= minHeight) {
+            height = elementRef.current.style.height = `${newHeight}px`;
+            translateY = initialTranslateY - (initialPointerY - clientY);
+          }
+          const newWidth = initialWidth + (initialPointerX - clientX);
+          if (newWidth >= minWidth) {
+            width = elementRef.current.style.width = `${newWidth}px`;
+            translateX = initialTranslateX - (initialPointerX - clientX);
+          }
+          elementRef.current.style.transform = `translate(${translateX}px, ${translateY}px)`;
+          break;
+        }
+        case 'top': {
+          const newHeight = initialHeight + (initialPointerY - clientY);
+          if (newHeight < minHeight) {
+            break;
+          }
+          height = elementRef.current.style.height = `${newHeight}px`;
+          translateY = initialTranslateY - (initialPointerY - clientY);
+          elementRef.current.style.transform = `translate(${translateX}px, ${translateY}px)`;
+          break;
+        }
+        case 'top-right': {
+          const newHeight = initialHeight + (initialPointerY - clientY);
+          if (newHeight >= minHeight) {
+            height = elementRef.current.style.height = `${newHeight}px`;
+            translateY = initialTranslateY - (initialPointerY - clientY);
+          }
+          width = elementRef.current.style.width = `${
+            initialWidth - (initialPointerX - clientX)
+          }px`;
+          elementRef.current.style.transform = `translate(${translateX}px, ${translateY}px)`;
+          break;
+        }
+        case 'right': {
```

Comment:
For right, bottom-right and bottom no need to check min-width and min-height since CSS handles it.